cnd/src/ethereum.rs

```diff
@@ -86,7 +204,70 @@ impl<T: Into<Vec<u8>>> From<T> for Bytes {
 #[cfg(test)]
 mod tests {
     use super::Log;
-    use crate::ethereum::TransactionReceipt;
+    use crate::ethereum::{Address, Hash, TransactionReceipt};
+    use serde::Deserialize;
+    use std::str::FromStr;
+
+    #[test]
+    fn deserialise_address() {
+        let json =
+            serde_json::Value::String("0xc5549e335b2786520f4c5d706c76c9ee69d0a028".to_owned());
+        let _: Address = Address::deserialize(&json).unwrap();
+    }
+
+    #[test]
+    #[should_panic]
```

Comment:
Note: Instead of `should_panic`, you could also test that `serde_json::from_value` returns an `Err` :)

Reply:
done